aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -423,6 +536,11 @@ inline Return Dispatcher::callWithDispatchKeySlowPath(const TypedOperatorHandle<
       } else {
         runRecordFunction(guard, op, dispatchKey);
       }
+      if (C10_UNLIKELY(guard.needsOutputs())) {
+        detail::ReturnValue<Return> result(kernel, op, dispatchKeySet, std::forward<Args>(args)...);
```

Comment:
This calls from an unboxed context, but this can call into either a boxed or an unboxed kernel. If you call into a boxed kernel, then the kernel will unbox the results for you before returning. And you'd be re-boxing them here. This is additional effort that we might be able to avoid. @ilia-cher did something similar to avoid the double boxing for the arguments.

Reply:
I have discussed with @ilia-cher. The new implementation with `push_outputs::copy` should handle this correctly.